lib/api/apiUtils/object/createAndStoreObject.js

```diff
@@ -70,7 +70,9 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
         return callback(err);
     }
 
-    const metaHeaders = isDeleteMarker ? [] :
+    const metaHeaders = objMD && isDeleteMarker ?
+        { 'x-amz-meta-scal-location-constraint':
+        objMD[constants.objectLocationConstraintHeader] } :
```

Comment:
This is mostly if someone put an object using the `x-amz-meta-scal-location-constraint` header instead of specifying the location constraint at the bucket level. If they create a delete marker by deleting that object, the location constraint is pulled from the existing object's metadata so the delete marker is created in the existing object's datastore ([line 123](https://github.com/scality/S3/pull/926/files/09b006811a5b2e03eb5549fdfd63f4d58228470f#diff-e45defa8c8c53634533cf165cc9232ceR123)).

I figured this is how it should work since the `x-amz-meta-scal-location-constraint` cannot be specified during a delete, but if we're going to abandon object level location constraints, this will be removed later on anyway.

Reply:
Let's get rid of the ?:?: 

Also, instead of messing with metaheaders, let's add to the request object, the x-amz-scal-location-constraint that might be stored on an object so that if it is a delete marker we can save that info with the delete marker.